cpp/src/parquet/column_writer.cc

```diff
@@ -1209,28 +1209,31 @@ Status ConvertDictionaryToDense(const ::arrow::Array& array, MemoryPool* pool,
   return Status::OK();
 }
 
-template <typename DType>
-class TypedColumnWriterImpl : public ColumnWriterImpl, public TypedColumnWriter<DType> {
+template <typename ParquetType>
```

Comment:
Since the implementation uses both parquet and arrow types, I thought it is better to be explicit. I can restore the previous name though.

Reply:
This makes sense and already used by some functions in this file.

Reply:
> Since the implementation uses both parquet and arrow types, I thought it is better to be explicit.

+1